realsense_camera/src/get_debug_info.cpp

```diff
@@ -0,0 +1,203 @@
+
+  if (!pipe)
+  {
+    throw std::runtime_error("popen(\"" + std::string(cmd) + "\", \"r\") failed");
+  }
+
+  while (!feof(pipe.get()))
+  {
+    if (fgets(buffer.data(), 128, pipe.get()) != NULL)
+    {
+      result += buffer.data();
+    }
+  }
+
+  std::string::size_type pos = result.find_last_not_of("\n \t");
+  if (pos != std::string::npos)
+  {
+    result = result.substr(0, pos + 1);
+  }
+  return result;
+}
+
+int main(int argc, char **argv)
+{
+  ros::init(argc, argv, "debug_tool");
+  ros::NodeHandle n;
+  std::string message = "\n----------CUT-PASTE-THIS----------\n";
+  message = message + "|Version | Your Configuration | \n|:----------|:----------| \n";
+
+  /* Acquire OS */
+  message = message + "|Operating System | ";
+  std::string operating_system;
+  try
+  {
+    operating_system = getSystemCallOutput("lsb_release -ds");
+    if (operating_system.empty())
+    {
+      operating_system = "Unable to detect operating system";
+    }
+  }
+  catch(const std::exception& e)
+  {
+    operating_system = e.what();
+  }
+  message = message + operating_system + "|\n";
+
+  /* Acquire Kernel Release */
+  struct utsname* buf = new utsname();
+  int uname_return = uname(buf);
+  std::string kernel;
+  if (uname_return == 0)
+  {
+    kernel = buf->release;
+  }
+  else
+  {
+    kernel = "Unable to detect Kernel";
+  }
+  message = message + "|Kernel | ";
+  message = message + kernel + "|\n";
+
+  /* Acquire ROS Version */
+  message = message + "|ROS | ";
+  std::string ros_version;
+  char* ros_version_char = getenv("ROS_DISTRO");
+  if (ros_version_char == NULL) ros_version = "Unable to detect ROS version";
+  else ros_version = ros_version_char;
+  message = message + ros_version + "| \n";
+
+  /* Acquire ROS RealSense Version */
+  message = message + "|ROS RealSense | ";
+  std::string realsense_version;
+  try
+  {
+    realsense_version = getSystemCallOutput("rosversion realsense_camera");
+    if (realsense_version.empty())
+    {
+      realsense_version = "Unable to detect realsense version";
+    }
+  }
+  catch(const std::exception& e)
+  {
+    realsense_version = e.what();
+  }
+  message = message + realsense_version + "|\n";
+
+  /* Acquire librealsense version */
+  message = message + "|librealsense | ";
+  std::string librealsense_version;
+  try
+  {
+    librealsense_version = getSystemCallOutput("rosversion librealsense");
+    if (librealsense_version.empty())
+    {
+      librealsense_version = "Unable to detect librealsense version";
+    }
+  }
+  catch(const std::exception& e)
+  {
+    librealsense_version = e.what();
+  }
+  message = message + librealsense_version + "|";
+
+  /* Acquire detected cameras' types and firmwares */
+  rs_context_ = rs_create_context(RS_API_VERSION, &rs_error_);
+
+  int num_of_cameras = rs_get_device_count(rs_context_, &rs_error_);
+
+  if (rs_error_)
+  {
+    message = message + "\n|Camera Type-Firmware | Error detecting cameras |";
+    rs_error_ = NULL;
+  }
+
```

Comment:
@reaganlo We will update the Issue template to have the users use this new tool instead of the manual commands once it is released through the ROS build farm.